### PR TITLE
CVE-2024-6387 use custom apt repo

### PR DESCRIPTION
**Pull request by MoteHue**
Draft as this Kayobe patch is needed: https://review.opendev.org/c/openstack/kayobe/+/923871
(Or we create an ugly post hook in the meantime)

### CODE REVIEW

**Review by markgoddard — COMMENTED** [review]
Could you add a reno?

File: etc/kayobe/stackhpc.yml [hunk 1]
```diff
@@ -47,6 +47,10 @@ stackhpc_repo_ubuntu_jammy_version: "{{ stackhpc_repo_distribution }}"
 stackhpc_repo_ubuntu_jammy_security_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/ubuntu/jammy-security/{{ stackhpc_repo_ubuntu_jammy_security_version }}"
 stackhpc_repo_ubuntu_jammy_security_version: "{{ stackhpc_repo_distribution }}"
 
+# Ubuntu jammy CVE-3034-6287
+stackhpc_repo_ubuntu_jammy_cve_2024_6387_url: "{{ stackhpc_repo_mirror_url }}/pulp/content/ubuntu-jammy-cve-2024-6387/{{ stackhpc_repo_ubuntu_jammy_cve_2024_6387_version }}"
+stackhpc_repo_ubuntu_jammy_cve_2024_6387_version: "{{ stackhpc_repo_distribution }}"
```
Comment (markgoddard) [inline etc/kayobe/stackhpc.yml:52]: We need an override for this in `stackhpc-ci.yml` in the ci-aio, ci-builder and ci-multinode envs.

Comment (MoteHue) [reply etc/kayobe/stackhpc.yml:52]: Done

File: etc/kayobe/environments/ci-aio/stackhpc-ci.yml [hunk 2]
```diff
@@ -50,6 +50,7 @@ stackhpc_repo_ubuntu_focal_version: "{{ stackhpc_pulp_repo_ubuntu_focal_version
 stackhpc_repo_ubuntu_focal_security_version: "{{ stackhpc_pulp_repo_ubuntu_focal_security_version }}"
 stackhpc_repo_ubuntu_jammy_version: "{{ stackhpc_pulp_repo_ubuntu_jammy_version }}"
 stackhpc_repo_ubuntu_jammy_security_version: "{{ stackhpc_pulp_repo_ubuntu_jammy_security_version }}"
+stackhpc_repo_ubuntu_jammy_cve_2024_6387_version: ""
```
Comment (markgoddard) [inline etc/kayobe/environments/ci-aio/stackhpc-ci.yml:53]: A bit weird, but I suppose it's because we didn't use a versioned distribution.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**